### PR TITLE
Update array_ops.py

### PR DESCRIPTION
**Pull request by nikochiko**
Update documentation, formatting and fix typos for several methods.

### CODE REVIEW

**Comment by nikochiko** [comment]
Batching fixes together. Will add commits on top of this.

File: tensorflow/python/ops/array_ops.py [hunk 1]
```diff
@@ -457,16 +465,23 @@ def broadcast_dynamic_shape(shape_x, shape_y):
...
   TensorShape whose value is [5, 2, 3].
 
   This is useful when validating the result of a broadcasting operation when the
   tensors have statically known shapes.
 
+  Example:
+
+  >>> shape_x = tf.TensorShape([1, 2, 3])
+  >>> shape_y = tf.TensorShape([5, 1 ,3])
+  >>> broadcast_static_shape(shape_x, shape_y)
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: ```>>> broadcast_static_shape(shape_x, shape_y)```

should be ```>>> tf.broadcast_static_shape(shape_x, shape_y)```

File: tensorflow/python/ops/array_ops.py [hunk 2]
```diff
@@ -1630,9 +1636,21 @@ def boolean_mask_v2(tensor, mask, axis=None, name="boolean_mask"):
   ragged tensors, and can be used if you need to preserve the masked dimensions
   of `tensor` (rather than flattening them, as `tf.boolean_mask` does).
 
+  Examples:
+
+  >>> tensor = [0, 1, 2, 3]  # 1-D example
+  >>> mask = np.array([True, False, True, False])
+  >>> boolean_mask(tensor, mask)
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: ```>>> boolean_mask(tensor, mask)```
should be 
```>>> tf.boolean_mask(tensor, mask)```
Do the same to all parts where you missed calling ```tf.``` before the function names

File: tensorflow/python/ops/array_ops.py [hunk 3]
```diff
@@ -3157,41 +3157,59 @@ def edit_distance(hypothesis, truth, normalize=True, name="edit_distance"):
   You can normalize the edit distance by length of `truth` by setting
   `normalize` to true.
 
-  For example, given the following input:
+  For example:
+
+  Given the following input,
+  * `hypothesis` is a `tf.SparseTensor` of shape `[2, 1, 1]`
```
Comment (steph-en-m) [inline tensorflow/python/ops/array_ops.py]: ```...shape `[2, 1, 1]` ```
tensor shapes are tuples not lists. change every occurrence similar to the above to ```...shape `(2, 1, 1)` ```

File: tensorflow/python/ops/check_ops.py [hunk 4]
```diff
@@ -2107,24 +2107,30 @@ def ensure_shape(x, shape, name=None):
...
   print(y.shape)
-  ==> TensorShape([Dimension(None), Dimension(3), Dimension(3)])
+  TensorShape([None, 3, 3])
 
-  with tf.compat.v1.Session() as sess:
-    # Raises tf.errors.InvalidArgumentError, because the shape (3,) is not
-    # compatible with the shape (None, 3, 3)
-    sess.run(y, feed_dict={x: [1, 2, 3]})
+  >>> with tf.compat.v1.Session() as sess:
+  >>> sess.run(y, feed_dict={x: [1, 2, 3]})
```
Comment (steph-en-m) [inline tensorflow/python/ops/check_ops.py]: Fix indentation here since `sess.run()` executes inside the `with` statement.

Comment (steph-en-m) [reply tensorflow/python/ops/check_ops.py]:
```
>>> with tf.compat.v1.Session() as sess:
...   sess.run(y, feed_dict={x: [1, 2, 3]})
```

**Comment by nikochiko** [comment]
@steph-en-m Looking good?

**Comment by steph-en-m** [comment]
Looks good to me. @k-w-w

**Comment by nikochiko** [comment]
Sorry, I had made this PR on `master`, so moved it to https://github.com/tensorflow/tensorflow/pull/34864. Please see.

**Pull request closed without merging**